test/node/double_tests.js

```diff
@@ -35,4 +36,74 @@ describe('Double', function () {
       });
     }
   });
+
+  describe('specialValues', () => {
+    function twiceSerialized(value) {
+      let serializedDouble = BSON.serialize({ d: value });
+      let deserializedDouble = BSON.deserialize(serializedDouble, { promoteValues: true });
+      let newVal = deserializedDouble.d;
+      return newVal;
+    }
+
+    it('inf', () => {
+      let value = Infinity;
+      let newVal = twiceSerialized(value);
+      expect(value).to.equal(newVal);
+    });
+
+    it('-inf', () => {
+      let value = -Infinity;
+      let newVal = twiceSerialized(value);
+      expect(value).to.equal(newVal);
+    });
+
+    it('NaN', () => {
+      let value = NaN;
+      let newVal = twiceSerialized(value);
+      expect(Number.isNaN(newVal)).to.equal(true);
+    });
+
+    it('NaN with payload', function () {
+      if (!isBrowser() && getNodeMajor() < 10) {
+        this.skip();
+      }
+      let buffer = Buffer.from('120000000000F87F', 'hex');
+
+      const dv = new DataView(buffer.buffer, buffer.byteOffset, buffer.byteLength);
+      let value = dv.getFloat64(0, true);
+
+      let serializedDouble = BSON.serialize({ d: value });
+      expect(serializedDouble.subarray(7, 15)).to.deep.equal(buffer);
+      let { d: newVal } = BSON.deserialize(serializedDouble, { promoteValues: true });
+      expect(Number.isNaN(newVal)).to.equal(true);
+    });
+
+    it('0', () => {
+      let value = 0;
+      let orig = new Double(value);
+      let newVal = twiceSerialized(orig);
+      expect(value).to.equal(newVal);
+    });
+
+    it('-0', () => {
+      let value = -0;
+      let orig = new Double(value);
+      let newVal = twiceSerialized(orig);
+      expect(Object.is(newVal, -0)).to.be.true;
+    });
+
+    // TODO (NODE-4335): -0 should be serialized as double
+    it.skip('-0 serializes as Double', () => {
+      let value = -0;
+      let serializedDouble = BSON.serialize({ d: value });
+      let type = serializedDouble[5];
+      expect(type).to.not.equal(BSON.BSON_DATA_NUMBER);
```

Comment:
this isn't asserting that `-0` serializes to Double, it would be better to have the actual logic that shows that `-0` is a double that is appropriately roundtripped here

Reply:
Got it, I've changed it to check that the current implementation has it equalling an int and not equalling a Double.